portable/Common/mpu_wrappers.c

```diff
@@ -1429,13 +1429,19 @@
     #if ( configSUPPORT_DYNAMIC_ALLOCATION == 1 )
         StreamBufferHandle_t MPU_xStreamBufferGenericCreate( size_t xBufferSizeBytes,
                                                              size_t xTriggerLevelBytes,
-                                                             BaseType_t xIsMessageBuffer ) /* FREERTOS_SYSTEM_CALL */
+                                                             BaseType_t xIsMessageBuffer,
+                                                             StreamBufferCallbackFunction_t pxSendCompletedCallback,
+                                                             StreamBufferCallbackFunction_t pxReceiveCompletedCallback ) /* FREERTOS_SYSTEM_CALL */
         {
             StreamBufferHandle_t xReturn;
             BaseType_t xRunningPrivileged;
 

```

Comment:
The current privilege level should be checked before allowing stream buffer creation with pxSendCompletedCallback or pxReceiveCompletedCallback set.

Otherwise code not meant to run as privileged may run as privileged, bypassing configENFORCE_SYSTEM_CALLS_FROM_KERNEL_ONLY.

Reply:
Alternatively, the callback could be invoked after vPortResetPrivilege.

Reply:
Made the feature currently disabled for MPU ports.